### ISSUE TITLE
Slowness when loading stocktakes and stocktake lines 

### ISSUE THREAD

**Issue by DhanyaHerath** [msupply-foundation/open-msupply#2904]
## What went wrong? 😲
 
We are seeing slowness when loading stocktakes and stocktake lines on android, in local scenarios with a small dataset. 
More details can be found [in this issue](https://github.com/msupply-foundation/open-msupply-internal/issues/41)

## Expected behaviour 🤔
Reasonable speeds for these smaller datasets. We shouldn't be seeing this much of a lag here 

## Your environment 🌱

 - Version: 1.6.1
 - Platform:
   - [x] android (tablet), Samsung A8
   - [ ] browser (extra points if you tell us which one)
   - [ ] desktop (windows)
   - [ ] desktop (macOS)
   - [ ] server (windows)

**Comment by mark-prins** [comment]
The query can be pretty slow (~300-500ms) and is being used in at least four separate places. I don't see the query being called that many times however, so it isn't simply a multiplication of the query delay. I think it worth refactoring all the same, and have raised a separate issue (#2915) to investigate api speed.

`useStocktakeSelector` is called 24 times when displaying the detail view!

**Comment by nisha-dangol** [comment]
Tested this in v1.7.0-rc1 apk as well as exe. 
I used a master-list with around 3k items, and the average time it took to load/save the item-line was around 4 seconds for me, which I think is okay given the no. of items it had. So, I call this a PASS! [Here's the reference recording](https://github.com/msupply-foundation/open-msupply/assets/59193490/51ffd866-1a16-4e22-9d9c-7475b0db43f6) to how fast it loads than in the issue mentioned!
